src/ZkFold/Cardano/UPLC/Rollup.hs

```diff
@@ -98,22 +101,27 @@ rollup (RollupSetup ledgerRules dataCurrency threadValue feeAddress) (UpdateRoll
     && sort update' == sort update
 
     -- Check the current rollup output
-    && val == threadValue
+    -- && val == threadValue  -- Doesn't work because 'val' also contains lovelaces
+    && member threadCurrency (getValue val)
 
     -- Check the next rollup output
-    && out' == TxOut addr threadValue (OutputDatum (Datum $ mkI state')) Nothing
+    -- && out' == TxOut addr threadValue (OutputDatum (Datum $ mkI state')) Nothing
```

Comment:
Need to check value of `out'` separately, because it contains lovelaces in addition to the thread token.

Reply:
Again, my idea was that `threadValue` already includes ada and the thread token. It is not clear which solution is more efficient.

Reply:
I see.  Will implement my idea, but will eventually benchmark to see which approach is more efficient.